packages/generators/commons/src/BaseGeneratorConfig.ts

```diff
@@ -0,0 +1,17 @@
+import { z } from "zod";
+
+// Common per-generator configuration flags
+export interface BaseGeneratorConfig {
+    clientClassName: string;
+    noOptionalProperties: boolean;
+    extraDependencies: Record<string, string>;
+    defaultTimeoutInSeconds: number | "infinity" | undefined;
+}
```

Comment:
can we just use the zod schema to export the type here? https://github.com/fern-api/fern/blob/292f4f10d5c8c48f626cc17710c51d8555f1b372/packages/cli/config-management/generators-configuration/src/schemas/GithubLicenseCustomSchema.ts#L7

Reply:
Nice yes can do